src/screens/Instructor/Offering/EditingPages/PlaylistEditing.jsx

```diff
@@ -0,0 +1,193 @@
+/**
+ * Edit Playlist Page
+ * sub page for creating or editing a playlist
+ */
+
+import React, { useState, useEffect } from 'react'
+// UI & Layouts
+import { Grid, Form, Input, Select, Message, Icon, Divider } from 'semantic-ui-react'
+import { GeneralModal, GeneralLoader } from '../../../../components'
+import { SaveButtons, EditButtons } from './Buttons'
+// Vars
+import { api, handleData, util } from '../../../../util'
+
+/**
+ * @param type 'new' for creating, 'id' for editing
+ * @param id   type is new: offeringId, type is id: playlistId
+ * @param history for goBack
+ */
+export function PlaylistEditing ({match: {params: {id, type}}, history}) {
+  // determine whether is going to create or edit a playlist
+  const isNew = type === 'new' 
+  const path = '??' // TBD
+
+  /** 
+   * playlist - the original playlist info whiling editing the playlist 
+   */
+  const [playlist, setPlaylist] = useState(api.initialData.initialPlaylist)
+  /**
+   * playlistInfo - the object for recording the inputs
+   */
+  const [playlistInfo, setPlaylistInfo] = useState(playlist)
+  const [loading, setLoading] = useState(true)
+
+  /**
+   * Used while editinf a playlist
+   * GET all the needed info based on the playlist Id
+   */
+  useEffect(()=> {
+    api.contentLoaded(500)
+    if (!isNew) {
+      // setLoading(true)
+      // api.getData(path, id)
+      // .then( response => {
+      //   setPlaylist(response.data)
+      //   setPlaylistInfo(response.data)
+      //   setLoading(false)
+      // })
+    }
+  }, [])
+
+  /**
+   * Functions for http requests
+   */
+  const callBacks = {
+    onCreate: function () {
+      if ( isNew ) playlistInfo.offeringId = id
+      // api.postData(path, playlistInfo, response => {})
+      console.log(playlistInfo)
+    },
+    onUpdate: function () {
+      var data = handleData.updateJson(playlistInfo, playlist)
+      data.id = id
+      console.log(data)
+      // api.updateData(path, data, () => this.onClose())
```

Comment:
Any reason this commented out?

Reply:
Cause currently I cant get the playlist name and offeringId, so I will update it later